regression/tests/studio/test_course_update.py

```diff
@@ -29,47 +29,102 @@ def setUp(self):
             self.course_info['run']
         )
         self.course_update_page.visit()
+        self.course_update_content_selector = '#course-update-list li' \
+                                              ' .post-preview .update-contents'
+        self.course_update_text = 'New update:{}'.format(uuid4().hex)
 
-    def test_course_update(self):
+    def tearDown(self):
         """
-        Verifies creation, editing and deletion of a course update
+        Delete the courses created.
+
+        It is called after every test run irrespective of
```

Comment:
Interesting. After going through the documentation my impression was that if setUp succeeds,  tearDown will be called whether there is error or not. From [documentation](https://docs.python.org/3.3/library/unittest.html#unittest.TestCase.tearDown) 
"Method called immediately after the test method has been called and the result recorded. This is called even if the test method raised an exception, so the implementation in subclasses may need to be particularly careful about checking internal state. Any exception raised by this method will be considered an error rather than a test failure. This method will only be called if the setUp() succeeds, regardless of the outcome of the test method. The default implementation does nothing."

I can also do it using addCleanup which I think will run all the time even if setUp fails.